<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TrainingSetter</class>
 <widget class="QWidget" name="TrainingSetter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>735</width>
    <height>442</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>训练设置</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="3" column="5">
    <widget class="QPushButton" name="pushButton_Execute">
     <property name="text">
      <string>执行</string>
     </property>
    </widget>
   </item>
   <item row="3" column="4">
    <widget class="QPushButton" name="pushButton_SaveToSolver">
     <property name="text">
      <string>保存</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QPushButton" name="pushButton_SolverBrowser">
     <property name="text">
      <string>浏览</string>
     </property>
    </widget>
   </item>
   <item row="3" column="3">
    <widget class="QPushButton" name="pushButton_ReadFormSolver">
     <property name="text">
      <string>读取</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>训练配置;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QLineEdit" name="lineEdit_solverFilePath"/>
   </item>
   <item row="7" column="0" colspan="7">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>训练配置详情</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="toolTip">
         <string>上一次的梯度值在本次梯度下降中的权重</string>
        </property>
        <property name="text">
         <string>momentum: </string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="lineEdit_momentum"/>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="label_13">
        <property name="toolTip">
         <string>每训练多少次进行一次网络快照</string>
        </property>
        <property name="text">
         <string>snapshot:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_11">
        <property name="toolTip">
         <string>每训练多少次在终端显示一次信息</string>
        </property>
        <property name="text">
         <string>display:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="toolTip">
         <string>测试数据需要多少次迭代能够用完,与batch_size之积是测试样本的总数</string>
        </property>
        <property name="text">
         <string>test_iter:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="toolTip">
         <string>每训练多少张图片进行一次测试</string>
        </property>
        <property name="statusTip">
         <string/>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
        <property name="text">
         <string>test_interval:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="lineEdit_test_iter"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="lineEdit_test_interval"/>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_12">
        <property name="toolTip">
         <string>最大迭代次数</string>
        </property>
        <property name="text">
         <string>max_iter:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QLineEdit" name="lineEdit_snapshot_prefix"/>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="lineEdit_weight_decay"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_7">
        <property name="toolTip">
         <string>权重衰减(防止过拟合)</string>
        </property>
        <property name="text">
         <string>weight_decay:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_15">
        <property name="toolTip">
         <string>运行模式</string>
        </property>
        <property name="text">
         <string>solver_mode:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QLineEdit" name="lineEdit_net"/>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="label_16">
        <property name="toolTip">
         <string>网络结构定义文件位置</string>
        </property>
        <property name="text">
         <string>net: </string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="comboBox_solver_mode">
        <item>
         <property name="text">
          <string>CPU</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GPU</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label_14">
        <property name="toolTip">
         <string>网络快照的保存位置</string>
        </property>
        <property name="text">
         <string>snapshot_prefix:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="4">
       <widget class="QPushButton" name="pushButton_NetBrowser">
        <property name="text">
         <string>浏览</string>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QPushButton" name="pushButton_SnapBrowser">
        <property name="text">
         <string>浏览</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3" colspan="2">
       <widget class="QLineEdit" name="lineEdit_snapshot"/>
      </item>
      <item row="1" column="3" colspan="2">
       <widget class="QLineEdit" name="lineEdit_max_iter"/>
      </item>
      <item row="0" column="3" colspan="2">
       <widget class="QLineEdit" name="lineEdit_display"/>
      </item>
      <item row="5" column="0" colspan="5">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>学习率策略</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_8">
           <property name="toolTip">
            <string>学习率衰减策略</string>
           </property>
           <property name="text">
            <string>lr_policy:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="lineEdit_stepsize"/>
         </item>
         <item row="0" column="1" colspan="5">
          <widget class="QComboBox" name="comboBox_lr_policy">
           <item>
            <property name="text">
             <string>fixed(恒定学习率)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>step(base_lr * gamma ^ (floor(iter / stepsize)))</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>exp(base_lr * gamma ^ iter)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>inv(base_lr * (1 + gamma * iter) ^ (- power))</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>multistep(根据stepvalue更新的step方法)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>poly(base_lr (1 - iter/max_iter) ^ (power))</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>sigmoid(base_lr ( 1/(1 + exp(-gamma * (iter - stepsize)))))</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QLineEdit" name="lineEdit_gamma"/>
         </item>
         <item row="2" column="4">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>power:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>gamma:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="lineEdit_base_lr"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>stepsize:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>stepvalue:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QLineEdit" name="lineEdit_stepvalue"/>
         </item>
         <item row="2" column="5">
          <widget class="QLineEdit" name="lineEdit_power"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_5">
           <property name="toolTip">
            <string>基础学习率</string>
           </property>
           <property name="text">
            <string>base_lr: </string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_SolverBrowser</sender>
   <signal>clicked()</signal>
   <receiver>TrainingSetter</receiver>
   <slot>BrowseSolver()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>567</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_SnapBrowser</sender>
   <signal>clicked()</signal>
   <receiver>TrainingSetter</receiver>
   <slot>BrowseSnapPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>729</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_NetBrowser</sender>
   <signal>clicked()</signal>
   <receiver>TrainingSetter</receiver>
   <slot>BrowseNet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>729</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_ReadFormSolver</sender>
   <signal>clicked()</signal>
   <receiver>TrainingSetter</receiver>
   <slot>ReadSolver()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>530</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>728</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_lr_policy</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>TrainingSetter</receiver>
   <slot>policyChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>257</y>
    </hint>
    <hint type="destinationlabel">
     <x>363</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Execute</sender>
   <signal>clicked()</signal>
   <receiver>TrainingSetter</receiver>
   <slot>ExecuteTraining()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>698</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>727</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_SaveToSolver</sender>
   <signal>clicked()</signal>
   <receiver>TrainingSetter</receiver>
   <slot>SaveSolver()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>612</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>728</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>BrowseSolver()</slot>
  <slot>BrowseSnapPath()</slot>
  <slot>BrowseNet()</slot>
  <slot>ReadSolver()</slot>
  <slot>SaveSolver()</slot>
  <slot>ExecuteTraining()</slot>
  <slot>policyChange()</slot>
 </slots>
</ui>
